### DISCUSSION TITLE
Simulate server-side channel failure

### DISCUSSION THREAD

**Discussion by lali1-betsson** [rabbitmq/rabbitmq-dotnet-client#1313] [Q&A] [answered]
Sometimes we experience unexpected channel closures, initiated by server.
Sometimes our client-side custom recovery logic fails to properly create another one.

So I'd like to (ideally programmatically) force a server-side channel close, so that I could test our custom recovery logic.
(Closing the whole connection manually in management console is a different scenario, I'd like just to kill a single channel in a working connection).
Any ideas how to do this?

**Comment by lukebakken** [comment]
In the future, there's no need to cross-post a question.

See my response here - https://github.com/rabbitmq/rabbitmq-server/discussions/7644#discussioncomment-5334870

**Comment by michaelklishin** [comment, marked as answer]
A client can [trigger a channel exception](https://rabbitmq.com/channels.html#error-handling) by first declaring, say, an exchange as transient and then as durable (under the same name).

As far as RabbitMQ is concerned, that will terminate the channel due to a protocol exception. The
client will get a notification, not sure if this is the scenario @lali1-betsson is looking for.

**Reply by lali1-betsson** [reply]
Seems to be a good idea, thanks
